lib/docker/artifact_image.js

```diff
@@ -15,6 +16,46 @@ import sleep from '../util/sleep';
 
 let debug = Debug('docker-worker:artifactImage');
 
+async function decompressLz4File(inputFile) {
+  let outputFile = path.join(path.dirname(inputFile), path.basename(inputFile, '.lz4'));
+  let proc = spawn('lz4', ['-d', inputFile, outputFile]);
+  let err = [];
+  proc.stderr.on('data', data => {
+    err.push(data);
+  });
+
+  await new Promise((accept) => {
+    proc.on('exit', accept);
+  });
+
+  if (proc.exitCode !== 0) {
+    throw new Error(
+        'Could not decompress image file.' +
+        `Exit Code: ${proc.exitCode} Errors: ${err.join('\n')}`);
+  }
+  return outputFile;
+}
```

Comment:
lines 23 to 36 could be isolated in a helper function and called by `decompressLz4File` and `decompressZstdFile`